docs/okeps/okep-5494-ovn-kubernetes-mcp-server.md

```diff
@@ -0,0 +1,740 @@
+  as they capture all network traffic on the host, requiring careful
+  consideration of privacy and performance impact while being chosen for
+  execution.
+* **Short-term Mitigation**: Strict command allowlisting exposing only read
+  operations (`ip route show`, `nft list ruleset`) while blocking modification
+  commands (`ip route add`, `nft add rule`)
+* **Long-term Path**: Requires RBAC-enabled wrapper tools from upstream
+  layered product teams (example netfilter, kernel networking)
+
+## Distributed Execution Context
+
+### **OVN-Kubernetes Architecture Challenge**
+
+In OVN-Kubernetes interconnect architecture, each node maintains its own local
+instances of critical databases and services:
+
+* **Northbound Database**: Local OVN northbound database per node
+* **Southbound Database**: Local OVN southbound database per node
+* **OpenVSwitch Database**: Node-specific OVS database and flow tables
+* **Host Networking**: Node-specific routing tables, conntrack zones, and
+  kernel state
+
+This distributed architecture means that troubleshooting commands must execute
+on the specific node where the relevant data resides.
+
+### **Node-Targeted Command Execution**
+
+**Node Selection Strategy**: All tools requiring node-specific data accept a
+`node_name` parameter as a required argument. The MCP server uses this
+parameter to:
+
+1. **Pod Selection**: Locate the appropriate `ovnkube-node` pod running on the
+   specified node using `pod.spec.nodeName` matching
+2. **Container Targeting**: Route OVN database commands to the correct
+   container within the ovnkube-node pod (e.g., `nb-ovsdb`, `sb-ovsdb`
+   containers)
+3. **Execution Context**: Execute host-level commands via
+   `oc debug node/&lt;node_name>` for direct host access
+
+**LLM Responsibility**: The LLM must determine the appropriate target node(s)
+based on the troubleshooting context:
+
+* **Pod-specific issues**: Use the node where the problematic pod is scheduled
+  (`kubectl get pod -o wide`)
+* **Network flow analysis**: Target nodes along the packet path (source node,
+  destination node, gateway nodes)
+* **Cluster-wide analysis**: Potentially execute commands across multiple
+  nodes for correlation
+
+We need to account for testing the LLM Responsibility side which is not
+something we can guarantee but something we are offloading to the LLM that we
+won't solve.
+
+## Deployment Strategy
+
+### **Flexible Deployment Modes**
+
+The MCP server is designed for flexible deployment without requiring elaborate
+cluster infrastructure. Multiple deployment modes support different use cases
+and security requirements:
+
+**CLI Tool Mode (Simplest)**:
+
+* Run the MCP server binary directly on a machine with `kubectl`/`oc` access
+  to the target cluster
+* Server uses existing cluster credentials and executes commands via standard
+  CLI tools
+* Suitable for both live cluster troubleshooting and offline
```

Comment:
Having kubectl isn't enough to interact with a `sos report` gathered data? I believe the MCP would have to be capable of translating commands to files with pre-recorded results.